src/pages/ShopPage/ShopPage.tsx

```diff
@@ -1,36 +1,70 @@
+import { FixedSizeGrid } from "react-window";
+import AutoSizer from "react-virtualized-auto-sizer";
 import { useEffect, useState } from "react";
-import ProductList from "./ProductList/ProductList";
-import Grid from "@mui/material/Unstable_Grid2";
+
+import GridMaterial from "@mui/material/Unstable_Grid2";
 import { Button } from "@mui/material";
-import { Product } from "../types";
-import Spinner from "../../components/Spinner/Spinner";
+import { Product, ViewType } from "../types";
 import { useFetch } from "../../hooks/useFetch";
-import { BACKEND_BASE_URL, PRODUCTS_PATH } from "../../constants";
+import Spinner from "../../components/Spinner/Spinner";
 import ErrorPanel from "../../components/ErrorPanel/ErrorPanel";
+import ProductItem from "./ProductItem/ProductItem";
+import { BACKEND_BASE_URL, PRODUCTS_PATH } from "../../constants";
+import { getScreenWidth, getDeviceViewSize, toMatrix } from "../../helpers";
 
-const limit = 10;
+import styles from "./ShopPage.module.scss";
 
 const ShopPage = () => {
-  const [products, setProducts] = useState<Product[]>([]);
+  const limit = 6;
+  const responsiveHeightRatio = 1.05;
+  let ignore = false;
+
+  const [products, setProducts] = useState<Product[][]>([]);
   const [hasMoreProducts, setHasMoreProducts] = useState(true);
   const [page, setPage] = useState(1);
+  const [columnCount, setColumnCount] = useState(3);
+  const [columnWidth, setColumnWidth] = useState(394);
+  const [rowHeight, setRowHeight] = useState(420);
```

Comment:
The same here, there should be no "magical numbers"
```suggestion
  const [columnCount, setColumnCount] = useState(DEFAULT_NUMBER_OF_COLUMNS);
  const [columnWidth, setColumnWidth] = useState(DEFAULT_COLUMN_WIDTH);
  const [rowHeight, setRowHeight] = useState(DEFAULT_ROW_HEIGHT);
```